src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/TransferCaseCallbackHandler.java

```diff
@@ -0,0 +1,120 @@
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final List<Role> ROLES = Collections.singletonList(CASEWORKER);
+    private static final List<CaseEvent> EVENTS = ImmutableList.of(TRANSFER_CASE);
+    private final BulkPrintTransferService bulkPrintTransferService;
+    private final CaseDetailsConverter caseDetailsConverter;
+
+    @Autowired
+    public TransferCaseCallbackHandler(
+        BulkPrintTransferService bulkPrintTransferService,
+        CaseDetailsConverter caseDetailsConverter
+    ) {
+        this.bulkPrintTransferService = bulkPrintTransferService;
+        this.caseDetailsConverter = caseDetailsConverter;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_START, this::prepopulateData,
+            CallbackType.ABOUT_TO_SUBMIT, this::performBulkPrintTransfer
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse prepopulateData(CallbackParams callbackParams) {
+
+     return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(ImmutableMap.of(
+                "dateOfTransfer", "TODAYS_DATE", // TODO
+                "reasonForTransfer",null,
+                "nameOfTransferCourt", null,
+                "addressOfTransferCourt", null
+            ))
+            .build();
+    }
+
+    private CallbackResponse performBulkPrintTransfer(CallbackParams callbackParams) {
+
+        CCDCase ccdCase = caseDetailsConverter.extractCCDCase(callbackParams.getRequest().getCaseDetails());
+
+        try {
+
+            var noticeOfTransferLetter = bulkPrintTransferService.generateNoticeOfTransferLetter(ccdCase);
+
+            bulkPrintTransferService.addNoticeOfTransferToCaseRecordToCaseDocuments(ccdCase, noticeOfTransferLetter);
+
+            if (!isDefendantLinked(ccdCase)) {
+                var transferNoticeLetter = bulkPrintTransferService.generateTransferNoticeLetter(ccdCase);
+
+                bulkPrintTransferService.addTransferLetterNoticeToCaseDocuments(ccdCase, transferNoticeLetter);
+            }
+
+            bulkPrintTransferService.sendCaseDocumentsToBulkPrint(ccdCase);
+
+            bulkPrintTransferService.sendClaimUpdatedEmailToClaimant(ccdCase);
+
+            if (isDefendantLinked(ccdCase)) {
+                bulkPrintTransferService.sendClaimUpdatedEmailToDefendant(ccdCase);
+            }
+
+        } catch (Exception e) {
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .errors(Collections.singletonList(
+                    "There was a technical problem. Some items may not have been sent. You may want to try again."))
+                .build();
+        }
```

Comment:
Probably best not to swallow this exception. CCD will display its own generic error message when there are exceptions in callbacks.

Reply:
I based the error handling on what I saw in GeneralLetterCallbackHandler.java. Please can you give me an example of the best practice.

Reply:
I would say that best practice would be to just allow the exception to cascade up. It would then report in the app insights monitoring and CCD will report a generic error message.

Reply:
Fixed in commit 4